fracmanager/fracmanager.go

```diff
@@ -298,16 +314,39 @@ func (fm *FracManager) Load(ctx context.Context) error {
 		}
 	}
 
-	if len(notSealed) == 0 {
-		fm.rotate()
-	} else {
-		if len(notSealed) > 1 {
-			logger.Info("sealing active fractions")
-			for _, active := range notSealed[:len(notSealed)-1] {
-				fm.seal(active)
-			}
+	if fm.active.ref == nil { // no active
+		_ = fm.rotate() // make new empty active
+	}
+
+	return nil
+}
+
+func (fm *FracManager) replayAll(ctx context.Context, actives []*frac.Active) error {
+	wg := sync.WaitGroup{}
+	defer wg.Wait()
+
+	for i, a := range actives {
+		if err := a.Replay(ctx); err != nil {
+			return err
+		}
+		if a.Info().DocsTotal == 0 {
+			a.Suicide() // remove empty
+			continue
+		}
+		r := fm.newActiveRef(a)
+		fm.fracs = append(fm.fracs, r.ref)
+
+		if i == len(actives)-1 { // last and not empty
+			fm.active = r
+		} else {
+			wg.Wait() // wait previous sealing complete
+
+			wg.Add(1)
+			go func() {
+				fm.seal(r)
+				wg.Done()
+			}()
```

Comment:
nit: maybe reduce nesting?
```suggestion
		if i == len(actives)-1 { // last and not empty
			fm.active = r
			continue
		}
		wg.Wait() // wait previous sealing complete

		wg.Add(1)
		go func() {
			fm.seal(r)
			wg.Done()
		}()
```